### PR TITLE
Add password confirmation feature and test

### LINKED ISSUE #20: Registration Page [Sad / Edge cases]
- [x] Passwords don't match 
- [x] Email is already registered

### PR DESCRIPTION
**Pull request by Diana20920**
# Modifies
- `app/controllers/users_controller.rb`:  added back :password_confirmation to our strong params
- `app/models/user.rb`: added validates_confirmation_of and removed commented out lines
- `app/views/users/new.html.erb`: added a break line before the form
- `spec/features/users/registration_spec.rb`: passing test for password confirmation

<details>
<summary>Issues</summary>
  closes #20 
<br>

</details>

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -21,6 +21,6 @@ def create
   private
 
   def user_params
-    params.require(:user).permit(:email, :password)
+    params.require(:user).permit(:email, :password, :password_confirmation)
```
Comment (arikalea) [inline app/controllers/users_controller.rb:24]: Nice job getting this to work!

File: spec/features/users/registration_spec.rb [hunk 2]
```diff
@@ -81,8 +81,7 @@
 
     click_on "Register"
 
-    expect(page).to have_content("Password confirmation failed, passwords must match. Please try again.
-")
+    expect(page).to have_content("Password confirmation doesn't match Password")
```
Comment (arikalea) [inline spec/features/users/registration_spec.rb:84]: This looks great!

**Review by arikalea — APPROVED** [review]
